test/udp_sender.js

```diff
@@ -249,4 +250,20 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', () => {
+        sender._client.on('error', err => {
+            assert.equal(err.message, 'getaddrinfo ENOTFOUND foo.bar.com')
```

Comment:
> Nit: this isn't really testing the new error handler you're adding. The test should simply try to send a span when the sender has the wrong agent address, and without the new handler it should throw uncaught error, but with the handler just log it to console and succeed.

```js
sender._host = 'foo.bar.com';
sender._port = 1234;
let reporter = new RemoteReporter(sender);
tracer = new Tracer(
    'test-service-name',
     reporter,
     new ConstSampler(true)
);
```
the above code (in my test case) is creating as sender that will attempt to send the span to a host that will throw an error on DNS lookup. now if I commented out the event handler in that test case, the test would fail and mirror the error that the issue stemmed from. How the same event emitter that is in the code is in my test and we expect that event emitter to be called which it is. 

The above reasoning is why I believe it is a valid test case: Here are some resources to back up my method for testing event emitters:
https://stackoverflow.com/questions/37232574/node-js-testing-if-event-was-emitted-using-sinon-js-without-the-use-of-timeouts